File: examples/sdk/python/composition.py
from mmos import MMOSClient


def main():

    client = MMOSClient(

        base_url="http://localhost:8080/api/v1",

        api_key="YOUR_API_KEY"
    )

    composition = client.compositions.create(

        name="AI Blog Production",

        description="Generate blog content using AI workflow",

        workflow="workflow.pipeline.multimedia-content",

        runtime={

            "provider": "openai",

            "model": "gpt-4.1"
        },

        input={

            "topic": "Artificial Intelligence",

            "language": "en"
        }
    )

    print(f"Composition ID: {composition.id}")


if __name__ == "__main__":
    main()

File: examples/sdk/typescript/composition.ts
import { MMOSClient } from "@mmos-ai/sdk";

async function main() {

    const client = new MMOSClient({

        baseUrl: "http://localhost:8080/api/v1",

        apiKey: "YOUR_API_KEY"
    });

    const composition = await client.compositions.create({

        name: "AI Blog Production",

        description: "Generate blog content using AI workflow",

        workflow: "workflow.pipeline.multimedia-content",

        runtime: {

            provider: "openai",

            model: "gpt-4.1"
        },

        input: {

            topic: "Artificial Intelligence",

            language: "en"
        }
    });

    console.log("Composition ID:", composition.id);
}

main().catch(console.error);